serverrpc/attach.go

```diff
@@ -22,23 +22,41 @@ func (s *ServerRPC) Attach(stream types.PublicAPI_AttachServer) error {
 	go func() {
 		for {
 			cmd, err := stream.Recv()
+			if err == io.EOF {
+				return
+			}
 			if err != nil {
+				glog.Errorf("Receive from stream error: %v", err)
 				return
 			}
-			if _, err := iw.Write(cmd.Data); err != nil {
+
+			n, err := iw.Write(cmd.Data)
+			if err != nil {
+				glog.Errorf("Write pipe error: %v", err)
 				return
 			}
+			if n != len(cmd.Data) {
+				glog.Errorf("Write data length is not enough, write: %d, success: %d", len(cmd.Data), n)
+			}
```

Comment:
Here you spot an error, but won't interrupt the stream. Is this what you want exactly?